autobahn/wamp/interfaces.py

```diff
@@ -262,7 +262,10 @@ class ISession(object):
     @abc.abstractmethod
     def onConnect(self):
         """
-        Callback fired when the transport this session will run over has been established.
+        Callback fired when the transport this session will run over has
+        been established.
+
```

Comment:
Yes, I believe so.

All I'm really changing in this PR is how the error-cases are handled. Currently, even with the ``try..catch`` and (recent) onUserError handler we still would simply never log an error *if* the handler happened to be async, and didn't have an error until after the first yield.

Now, we will always attach _swallow_error() which will at least log the error (and then ignore it). Of course, this would be a "last-chance" handler -- if user-code put their own error handling inside the onJoin or whatever, they would get called first.

At least, that's the theory ;)

I did add both "returns Deferred" and "simply throws" unit-tests for all the ISession handlers, *and* covered all the code-paths in protocol.py that lead to those handlers being called.

So: I don't expect *any* user-visible changes from this PR, except that errors now get logged.